pkg/resource/managed_reconciler.go

```diff
@@ -417,6 +421,23 @@ func (r *ManagedReconciler) Reconcile(req reconcile.Request) (reconcile.Result,
 	}
 
 	if !observation.ResourceExists {
+		if !IsConditionTrue(managed.GetCondition(v1alpha1.TypeReferencesResolved)) {
+			if err := r.managed.ResolveReferences(ctx, managed); err != nil {
+				// update the status according to the type of the err
+				switch err.(type) {
+				case NotReadyError:
```

Comment:
Good to know. Are you suggesting to use a different name?

Reply:
It's actually because of too much similarity with `error` type. If `NotReadyError` interface has a method other than `Error() string`, that'd be sufficiently different I think. It'd still satisfy `Error` interface and others wouldn't satisfy `NotReadyError` since they don't know about that other method; maybe a method to get list of unresolved dependencies? https://github.com/crossplaneio/crossplane-runtime/pull/46#discussion_r335495121

Reply:
@muvaf I have changed the `NotReadyError` in the later commits, let me know what you think